client/adventureMap/CInfoBar.cpp

```diff
@@ -266,13 +322,144 @@ void CInfoBar::showDate()
 	redraw();
 }
 
-void CInfoBar::showComponent(const Component & comp, std::string message)
+
+void CInfoBar::pushComponents(const std::vector<Component> & components, std::string message, int timer)
+{
+	auto actualPush = [&](const std::vector<Component> & components, std::string message, int timer, size_t max){
+		std::vector<Component> vect = components; //I do not know currently how to avoid copy here
+		while(!vect.empty())
```

Comment:
1) Do you need lambda here? It is called once and is not executed in async mode at some point later
2) Personal opinion, but I would rather call it "performPush", "pushImpl", "doPush". For me "actualPush" does not sounds like a good name for some reason...
3) You worry about "optimization" too much. Oh noes, you've copied vector with 5? 10? elements which took 1 mks. But the you need to render this components which can easily take 100 or even 1000 times more than this copy. There is no point in such micro-optimizations. If you want to improve performance the you need to investigate actual timings via profiler.

Reply:
1. These lambda is purely for readability. Of course, it can be changed to not using lambda, but what's wrong with it.
2. Maybe. It can be called pushImpl, for sure. Can be renamed later.
3. On huge pandoras this vector can contain around 1000 elements.

Reply:
@IvanSavenko should I change anything here?

Reply:
Up to you.
In my opinion lambda is unnecessary, but this is very minor issue, feel free to ignore.